src/nvim/edit.c

```diff
@@ -2992,15 +2999,10 @@ static void ins_compl_addleader(int c)
   if (ins_compl_need_restart())
     ins_compl_restart();
 
-  /* When 'always' is set, don't reset compl_leader. While completing,
-   * cursor doesn't point original position, changing compl_leader would
-   * break redo. */
-  if (!compl_opt_refresh_always) {
-    xfree(compl_leader);
-    compl_leader = vim_strnsave(get_cursor_line_ptr() + compl_col,
-        (int)(curwin->w_cursor.col - compl_col));
-    ins_compl_new_leader();
-  }
+  xfree(compl_leader);
+  compl_leader = vim_strnsave(get_cursor_line_ptr() + compl_col,
```

Comment:
Another change is when typed char, call the ins_compl_new_leader even refresh_always is setted. So complete func can immediately get called after change leader char. Instead of have to waiting for CursorMoved event and reinvoke complete function by feedkeys, which also cause flick.
I had tested and redo seems fine